compute-nodes.yaml

```
@@ -0,0 +1,94 @@
+- import_playbook: common.yaml
+
+- hosts: all
+  gather_facts: no
+
+  tasks:
+  # Build a list of worker nodes with indexes
+  - name: Build worker list
+    set_fact:       
+      worker_list: "{{ worker_list | default([]) + [ item.1 + '-' + item.0 | string ] }}"
+    with_indexed_items: "{{ [os_port_worker] * os_compute_nodes_number }}"
+
+  # Create the SDN/primary port for each worker node
+  - name: 'Create the Compute ports'
+    os_port:
+      name: "{{ item }}"
+      network: "{{ os_network }}"
+      security_groups:
+      - "{{ os_sg_worker }}"
+      allowed_address_pairs:
+      - ip_address: "{{ os_ingressVIP }}"
+    with_items: "{{ worker_list }}"
+    register: ports
+
+  # Tag each SDN/primary port with cluster name
+  - name: 'Set Compute ports tag'
+    command:
+      cmd: "openstack port set --tag {{ cluster_id_tag }} {{ item }}"
+    with_items: "{{ worker_list }}"  
```

`with_items`is deprecated in Ansible, we should use `loop`.
See this [transition doc](https://docs.ansible.com/ansible/latest/user_guide/playbooks_loops.html#with-items).

```suggestion
  - name: 'Set Compute ports tag'
    command:
      cmd: "openstack port set --tag {{ cluster_id_tag }} {{ item }}"
    loop: "{{ worker_list }}"  
```

---

Doh... my original code had used loop: :).  Should I go back to that?  It is much cleaner.

---

This is also from our UPI scripts :). Unless we fix it in UPI, we should not change that here for now